src/main/java/com/google/sps/servlets/PreviousTerms.java

```diff
@@ -0,0 +1,112 @@
+package com.google.sps.servlets;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/prev-terms")
+public class PreviousTerms extends HttpServlet {
+  private DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<Entity> prevTermsData = getPreviousTerms(db, request);
+    String prevTermsDataJSON = makeJSON(prevTermsData);
+    response.setContentType("application/json;");
+    response.getWriter().println(prevTermsDataJSON);
+  }
+
+  private List<Entity> getPreviousTerms(DatastoreService db, HttpServletRequest request) {
+    String schoolName = request.getParameter("school-name");
+    String courseName = request.getParameter("course-name");
```

Comment:
Yes so I discussed this with Nina today and we think it is feasible to make this change. Laura was not able to make our standup but we will discuss with her to see what impact this will have on her code as well, but based on the discuss Nina and I had we think it can be done with minor adjustments.

Reply:
Sounds good